File: UINavigationExtensionDemo/Base/BaseViewController.m
//
//  BaseViewController.m
//  UINavigationExtensionDemo
//
//  Created by lidan on 2020/10/29.
//

#import "BaseViewController.h"
#import "UIColor+RandomColor.h"

@interface BaseViewController ()

@end

@implementation BaseViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = self.randomColor;
}

#pragma mark - Getter

- (UIColor *)randomColor {
    if (!_randomColor) {
        _randomColor = [UIColor randomColor];
    }
    return _randomColor;
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Common/Comtrollers/BaseViewController.swift
//
//  BaseViewController.swift
//  NXNavigationExtensionSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class BaseViewController: UIViewController {

    private(set) var randomColor: UIColor?
    
    private lazy var lightColor: UIColor = UIColor.randomLight
    private lazy var darkColor: UIColor = UIColor.randomDark
    
    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = .customBackground
        
        randomColor = UIColor.customDynamic(lightMode: { [weak self] in
            guard let self = self else { return UIColor.randomLight }
            return self.lightColor
        }, darkMode: {  [weak self] in
            guard let self = self else { return UIColor.randomDark }
            return self.darkColor
        })
    }
    
    override var nx_titleTextAttributes: [NSAttributedString.Key : Any]? {
        return [NSAttributedString.Key.foregroundColor: UIColor.customTitle]
    }
    
    override var nx_barTintColor: UIColor? {
        return .customTitle
    }
    
    override var nx_shadowImageTintColor: UIColor? {
        return UIColor.customDynamic { UIColor.lightGray } darkMode: { UIColor.lightGray.withAlphaComponent(0.65) }
    }

}
